classes/submissionFile/Collector.inc.php

```diff
@@ -0,0 +1,253 @@
+    protected $assocIds = null;
+
+    /** @var boolean include submission files in the SUBMISSION_FILE_DEPENDENT stage */
+    protected $includeDependentFiles = false;
+
+    /** @var array get submission files matching one or more uploader users id */
+    protected $uploaderUserIds = null;
+
+    /** @var int */
+    public $count;
+
+    /** @var int */
+    public $offset;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Set fileStages filter
+     */
+    public function filterByFileStages(?array $fileStages): self
+    {
+        $this->fileStages = $fileStages;
+
+        return $this;
+    }
+
+    /**
+     * Set genreIds filter
+     */
+    public function filterByGenreIds(?array $genreIds): self
+    {
+        $this->genreIds = $genreIds;
+
+        return $this;
+    }
+
+    /**
+     * Set review rounds filter
+     */
+    public function filterByReviewRoundIds(?array $reviewRoundIds): self
+    {
+        $this->reviewRoundIds = $reviewRoundIds;
+
+        return $this;
+    }
+
+    /**
+     * Set review assignments filter
+     */
+    public function filterByReviewIds(?array $reviewIds): self
+    {
+        $this->reviewIds = $reviewIds;
+
+        return $this;
+    }
+
+    /**
+     * Set submissionIds filter
+     */
+    public function filterBySubmissionIds(?array $submissionIds): self
+    {
+        $this->submissionIds = $submissionIds;
+
+        return $this;
+    }
+
+    /**
+     * Set fileIds filter
+     */
+    public function filterByFileIds(?array $fileIds): self
+    {
+        $this->fileIds = $fileIds;
+
+        return $this;
+    }
+
+    /**
+     * Set assocType and assocId filters
+     *
+     * @param null|array $assocTypes One or more of the ASSOC_TYPE_ constants
+     * @param null|array $assocIds Match with ids for these assoc types
+     */
+    public function filterByAssoc(?array $assocTypes, ?array $assocIds = []): self
+    {
+        $this->assocTypes = $assocTypes;
+
+        if ($assocIds !== []) {
+            $this->assocIds = $assocIds;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Set uploaderUserIds filter
+     */
+    public function filterByUploaderUserIds(?array $uploaderUserIds): self
+    {
+        $this->uploaderUserIds = $uploaderUserIds;
+
+        return $this;
+    }
+
+    /**
+     * Whether or not to include dependent files in the results
+     */
+    public function filterByIncludeDependentFiles(bool $includeDependentFiles): self
+    {
+        $this->includeDependentFiles = $includeDependentFiles;
+
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     */
+    public function limit(int $count): self
+    {
+        $this->count = $count;
+
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     */
+    public function offset(int $offset): self
```

Comment:
Should be `?offset`

Reply:
No, this should return the Collector instance itself, to be chained with other methods if necessary.

Reply:
I mean `?int $offset` (as above: https://github.com/pkp/pkp-lib/pull/7166#discussion_r721708330)

Reply:
On lines 241 and 245 I'm counting if those values are greater than 0 to apply the query restraints.